app/javascript/components/provider-form/helper.js

```diff
@@ -0,0 +1,42 @@
+/** Function to add a custom-validator which checks the following criterias-
+ * - Field is required.
+ * - Field dose not contains trailing white spaces.
+*/
+export const providerValidatorMapper = {
+  trailingSpaces:
+    () =>
+      (value) => {
+        const regex = /\s+$/;
+        if (!value) {
+          return __('Required');
+        }
+        if (regex.test(value)) {
+          return __('Ensure there are no trailing spaces');
+        }
+        return true;
+      },
+};
+
+/** Function to update the form schema.
+ * If schema contains the field 'endpoints.default.hostname', then,
+ * replace its 'validate' with the custom-validate named 'trailingSpaces' .
+ */
+export const updateFieldValidator = (data) => {
+  const keys = Object.keys(data);
+  const fieldId = 'endpoints.default.hostname';
+
+  const item = keys.find((key) => {
+    const value = data[key];
+
+    if (value === fieldId) {
+      data.validate = [{ type: 'trailingSpaces' }];
+      return true;
+    } if (typeof value === 'object' && value !== null) {
+      return updateFieldValidator(value) !== undefined;
+    }
+
+    return false;
+  });
+
+  return item;
+};
```

Comment:
I have questions like -

1. should we apply the fix for all required fields,
2. or, should we apply it for all required text-fields where the user types the data?
3. Or if the custom-validator is the right solution to the issue.
4. Should we simply add the` [{ type: 'trailingSpaces' }]` in the api response. if so where all should we add them